From a45f072c5aaf7721647b83c46d5c4bc689ed71ba Mon Sep 17 00:00:00 2001
From: Wolfgang Grandegger <wg@grandegger.com>
Date: Thu, 18 May 2023 11:48:35 +0200
Subject: [PATCH 1/2] u-boot: add an option to generate u-boot-nodtb.bin

u-boot-nodtb.bin and u-boot.dtb is for example required to build
a fip image with recent versions of Arm Trusted Firmware.

Signed-off-by: Wolfgang Grandegger <wg@grandegger.com>
---
 boot/uboot/Config.in | 3 +++
 boot/uboot/uboot.mk  | 5 +++++
 2 files changed, 8 insertions(+)

diff --git a/boot/uboot/Config.in b/boot/uboot/Config.in
index f5c20f5168..abbc0085ce 100644
--- a/boot/uboot/Config.in
+++ b/boot/uboot/Config.in
@@ -281,6 +281,9 @@ config BR2_TARGET_UBOOT_FORMAT_DTB
 config BR2_TARGET_UBOOT_FORMAT_DTB_BIN
 	bool "u-boot-dtb.bin"
 
+config BR2_TARGET_UBOOT_FORMAT_NODTB_BIN
+	bool "u-boot-nodtb.bin"
+
 config BR2_TARGET_UBOOT_FORMAT_NAND_BIN
 	bool "u-boot-nand.bin"
 
diff --git a/boot/uboot/uboot.mk b/boot/uboot/uboot.mk
index 4eae8e95c3..127662afaa 100644
--- a/boot/uboot/uboot.mk
+++ b/boot/uboot/uboot.mk
@@ -96,6 +96,11 @@ UBOOT_BINS += u-boot-dtb.bin
 UBOOT_MAKE_TARGET += u-boot-dtb.bin
 endif
 
+ifeq ($(BR2_TARGET_UBOOT_FORMAT_NODTB_BIN),y)
+UBOOT_BINS += u-boot-nodtb.bin
+UBOOT_MAKE_TARGET += u-boot-nodtb.bin
+endif
+
 ifeq ($(BR2_TARGET_UBOOT_FORMAT_IMG),y)
 UBOOT_BINS += u-boot.img
 UBOOT_MAKE_TARGET += u-boot.img
-- 
2.34.1

